src/interval.jl

```diff
@@ -84,6 +84,36 @@ end
 
 Base.copy(x::Interval{T}) where T = Interval{T}(x.first, x.last, x.inclusivity)
 
+function Base.merge(a::AbstractInterval{T}, b::AbstractInterval{T}) where T
+    !overlapscontiguous(a,b) && throw(ArgumentError("$a and $b are not touching."))
+
+    left = min(LeftEndpoint(a), LeftEndpoint(b))
+    right = max(RightEndpoint(a), RightEndpoint(b))
+    return Interval(
+        left.endpoint,
+        right.endpoint,
+        Inclusivity(left.included, right.included)
+    )
+end
+
+function overlapscontiguous(a::AbstractInterval, b::AbstractInterval)
+    return overlaps(a,b) || contiguous(a,b)
+end
+
+function overlaps(a::AbstractInterval, b::AbstractInterval)
+    left = max(LeftEndpoint(a), LeftEndpoint(b))
+    right = min(RightEndpoint(a), RightEndpoint(b))
+
+    return left <= right
+end
+
+function contiguous(a::AbstractInterval, b::AbstractInterval)
+    left = max(LeftEndpoint(a), LeftEndpoint(b))
+    right = min(RightEndpoint(a), RightEndpoint(b))
+
+    return right.endpoint == left.endpoint && (left.included || right.included)
+end
```

Comment:
Both `overlaps` and `contiguous` should be added to the "tests/comparisons.jl" gauntlet